### PR TITLE
Remove ShareProtocol::None from the ShareProtocol enum

### PR DESCRIPTION
**Pull request by blaisedias**
The share protocol must now be specified always, and cannot be None.
Selecting the share protocol as nbd will have the same effect as,
selecting none previously.
Modify mocha test scripts to match these requirements.

In nexus test script, remove two test cases added around testing of
share value handling by mayastor
    1) testing ShareProtocol::None handling
    2) testing invalid value of share protocol
        This because some javascript library converts invalid values
        to 0 :-(. This can and has been tested with a hacked version of mctl.

Use std::Option<ShareProtocol> instead of ShareProtcol:None to indicate
unset share protocol in the nexus.

All invalid values of share protocol for the frontend/export purpose,
are now be handled intrinsically at the nexus rpc handling code,
so remove check in nexus share code.

### CODE REVIEW

**Review by jamie-0 — APPROVED** [review]
LGTM

A Jira is needed for the commented out test though, otherwise we'll forget about it.

**Comment by blaisedias** [comment]
> LGTM
> 
> A Jira is needed for the commented out test though, otherwise we'll forget about it.

That test was added by me under by the cas-23 set of changes. 
So, not worse than before w.r.t. main mayastor repo

File: csi/src/client.rs [hunk 1]
```diff
@@ -16,11 +16,10 @@ use rpc::service::mayastor_client::MayastorClient;
 
 fn parse_share_protocol(pcol: Option<&str>) -> Result<i32, Status> {
     match pcol {
-        None => Ok(rpc::mayastor::ShareProtocol::None as i32),
+        Some("nbd") => Ok(rpc::mayastor::ShareProtocol::Nbd as i32),
```
Comment (chriswldenyer) [inline csi/src/client.rs]: parse_share_protocol() looks like it is called only by create_replica() and share_replica(). 
Replicas use the back-end interface which use only iscsi and nvmf. Look at mayastor/src/replica.rs. 
nbd looks to like it should still be a invalid option.

Comment (blaisedias) [reply csi/src/client.rs]: As was None, so no change there.

Comment (blaisedias) [reply csi/src/client.rs]: Actually looking at the use cases
`$ find csi -name \*.rs -exec fgrep -Hn parse_share_protocol {} \;
csi/src/client.rs:17:fn parse_share_protocol(pcol: Option<&str>) -> Result<i32, Status> {
csi/src/client.rs:132:    let share = parse_share_protocol(matches.value_of("protocol"))?;
csi/src/client.rs:176:    let share = parse_share_protocol(matches.value_of("PROTOCOL"))?;
`
I think it the change is correct.
The underlying calls will fail if the share protocol is not supported by replicas.
The csi code should not make any assumption about the behaviour of replicas.

**Pull request closed without merging**